docs/tutorial/frontend-tut.md

```diff
@@ -99,42 +122,43 @@ component main() -> {
 
 The [cells section][lf-cells] instantiates hardware units like adders, memories and registers.
 The [wires section][lf-wires] contains [groups][lf-groups] that connect
-together hardware instances to perform some logical task such as incrementing a specific register.
+hardware instances to perform some logical task (e.g, incrementing a register).
 Finally, the [control section][lf-control] *schedules* the execution of groups using control operators such as `seq`, `par`, and `while`.
 
-We perform syntax-directed compilation by walking over nodes in the above AST and generating `cells`, `wires`, and `control` operations.
+We perform syntax-directed compilation by walking over the nodes of the AST and generating `cells`, `wires`, and `control` operations.
 
-#### Calyx Embedded DSL
 
-To make it easy to generate the hardware, we'll use Calyx's [`builder` module][builder-ex] in Python:
+### Calyx-Embedded DSL
+
+To make it easy to generate hardware, we'll use Calyx's [`builder` module][builder-ex] written in Python:
 ```python
 import calyx.builder as cb
 
 prog = cb.Builder() # A Calyx program
 main = prog.component("main") # Create a component named "main"
 ```
 
-#### `Decl` nodes
+
+### `Decl` nodes
 
```

Comment:
Maybe this is a good place to show what the `builder` module code will look like and what kind of Calyx it generates. Otherwise it sounds like we wrote the Calyx by hand.